assets/src/stories-editor/components/font-family-picker/index.js

```diff
@@ -23,50 +26,76 @@ function FontFamilyPicker( {
 	fonts = [],
 	onChange = () => {},
 	value = '',
+	id = '',
 } ) {
-	const defaultOption = {
-		value: '',
-		label: __( 'None', 'amp' ),
+	const results = fonts;
+	const suggest = ( query, syncResults ) => {
+		const searchResults = query ? results.filter( function( result ) {
+			return result.name.toLowerCase().indexOf( query.toLowerCase() ) !== -1;
+		} ) :
+			[];
+		syncResults( searchResults );
 	};
 
-	const options = fonts.map( ( font ) => ( {
-		value: font.name,
-		label: font.name,
-	} ) );
+	const suggestionTemplate = ( result ) => {
+		maybeEnqueueFontStyle( result.name );
+		const fallbacks = ( result.fallbacks ) ? ', ' + result.fallbacks.join( ', ' ) : '';
+		return result && `<span style='font-family: ${ result.name }${ fallbacks }'>${ result.name }</span>`;
+	};
 
-	const fontLabel = ( familyName ) => AMP_STORY_FONT_IMAGES[ familyName ] ?
-		AMP_STORY_FONT_IMAGES[ familyName ]( { height: 13 } ) :
-		familyName;
+	const inputValueTemplate = ( result ) => {
+		return result && result.name;
+	};
 
 	return (
-		<PreviewPicker
-			value={ value }
-			options={ options }
-			defaultOption={ defaultOption }
-			onChange={ ( { value: selectedValue } ) => onChange( '' === selectedValue ? undefined : selectedValue ) }
+		<BaseControl
 			label={ __( 'Font Family', 'amp' ) }
-			id="amp-stories-font-family-picker"
-			ariaLabel={ ( currentOption ) => {
-				return sprintf(
-					/* translators: %s: font name */
-					__( 'Font Family: %s', 'amp' ),
-					currentOption.label
-				);
-			} }
-			renderToggle={ ( { label } ) => fontLabel( label ) }
-			renderOption={ ( option ) => {
-				return (
-					<span className="components-preview-picker__dropdown-label" data-font-family={ option.value === '' ? undefined : option.value }>
-						{ fontLabel( option.label ) }
-					</span>
-				);
-			} }
-		/>
+			id={ id }
+			help={ __( 'Type to search for fonts', 'amp' ) }
+		>
+			<Autocomplete
+				id={ id }
+				source={ suggest }
+				templates={
+					{
+						suggestion: suggestionTemplate,
+						inputValue: inputValueTemplate,
+					}
+				}
+				minLength={ 2 }
+				onConfirm={ onChange }
+				showAllValues={ false }
+				confirmOnBlur={ false }
+				defaultValue={ value }
+				dropdownArrow={ () => '' }
+				preserveNullOptions={ true }
+				placeholder={ __( 'None', 'amp' ) }
+				displayMenu="overlay"
+				tNoResults={ () =>
+					__( 'No font found', 'amp' )
+				}
+				tStatusQueryTooShort={ ( minQueryLength ) =>
+					// translators: %d: the number characters required to initiate an author search.
```

Comment:
This translator hint is a bit misleading, as it's a font search, not an author search. It's not super important, just weird.